src/wallet3/db_schema.cpp

```diff
@@ -420,5 +425,41 @@ namespace wallet
   {
     return prepared_get<int64_t>("SELECT output_count FROM metadata WHERE id=0;");
   }
+  void
+  WalletDB::save_keys(const std::string& spend_priv_str, const std::string& spend_pub_str, const std::string& view_priv_str, const std::string& view_pub_str)
+  {
+    std::string query = "SELECT spend_priv, spend_pub, view_priv, view_pub FROM metadata WHERE id=0";
+
+    auto st = prepared_st(query);
+    st->executeStep();
+    auto from_db = db::get<std::string, std::string, std::string, std::string>(st);
+    const auto loaded_spend_priv_str = std::get<0>(from_db);
+    const auto loaded_spend_pub_str = std::get<1>(from_db);
+    const auto loaded_view_priv_str = std::get<2>(from_db);
+    const auto loaded_view_pub_str = std::get<3>(from_db);
+
+    if ((loaded_spend_priv_str != "" && loaded_spend_priv_str != spend_priv_str) ||
+        (loaded_spend_pub_str != "" && loaded_spend_pub_str != spend_pub_str) ||
+        (loaded_view_priv_str != "" && loaded_view_priv_str != view_priv_str) ||
+        (loaded_view_pub_str != "" && loaded_view_pub_str != view_pub_str))
+          throw std::runtime_error("loaded keys do not match database file");
```

Comment:
this is an ambiguous error as you've named the keys loaded *from* the database in this function `loaded_x_str` but refer to the keys passed into the function as "loaded" here.